src/mtcp/mtcp_restart.c

```diff
@@ -692,6 +692,23 @@ compute_vdso_vvar_addr(RestoreInfo *rinfo)
   mtcp_sys_close(mapsfd);
 }
 
+NO_OPTIMIZE
+static void
+unmap_one_memory_area(Area *area, int mapsfd)
```

Comment:
Maybe call this `unmap_one_memory_area_and_reread`.

See my other comment, about documenting the reason for re-reading.